plugins/inventory/vmware_vm_inventory.py

```diff
@@ -620,7 +643,7 @@ def parse_vim_property(vim_prop):
     elif isinstance(vim_prop, list):
         return [parse_vim_property(x) for x in vim_prop]
 
-    elif prop_type in ['bool', 'int', 'NoneType']:
+    elif prop_type in ['bool', 'int', 'NoneType', 'dict']:
```

Comment:
I'm not sure what is the use case of `dict` or what property have it. But it might be better to iterate each pair and parse it's value to be safe:

```py
vim_prop [key] = parse_vim_property(value)
```

Reply:
The new tag and category releationship data structure is `dict`, like

```
tag_category = {
    "OS": [ "Linux" ],
    "Backup": [ "Monday"],
}
```
where OS, Backup are Categories and "Linux" and "Monday" are tags associated with those respective categories.

I am OK using `parse_vim_property` for that. Let me know.

Reply:
That's OK and LGTM